internal/scylla/client.go

```diff
@@ -140,12 +144,34 @@ func (c *Client) doHttpRequestWithRetries(req *http.Request, retries int, retryB
 		if err == nil {
 			_ = resp.Body.Close() // We want to retry anyway.
 		}
+		sleepBeforeRetry(resp.Header.Get("Retry-After"), retryBackoffDuration)
 		return c.doHttpRequestWithRetries(req, retries-1, retryBackoffDuration*2)
 	}
 
 	return resp, err
 }
 
+func sleepBeforeRetry(timeToWait string, retryBackoffDuration time.Duration) {
+	var timeToSleep time.Duration
+	deadline, timeErr := time.Parse(time.RFC1123, timeToWait)
+	if timeErr == nil {
+		timeToSleep = time.Until(deadline)
+	} else {
+		seconds, _ := strconv.Atoi(timeToWait)
+		timeToSleep = time.Second * time.Duration(seconds)
+	}
+	if timeToSleep == 0 {
+		timeToSleep = retryBackoffDuration
+	}
+	switch {
+	case timeToSleep < minTimeToSleep:
+		timeToSleep = minTimeToSleep
+	case timeToSleep > maxTimeToSleep:
+		timeToSleep = maxTimeToSleep
+	}
+	time.Sleep(timeToSleep)
+}
+
```

Comment:
I am not sure why you bother to parse the Retry-After header, when you still don't respect it (you cut Retry-After to [100ms, 5s] if it exceeds). The Retry-After is here so the client can sleep precise amount of time instead of running loops. If you decide to not respect the Retry-After value, then it's better to not bother to parse it, just fix the sleep to 5s. Less code to maintain.

Alternatively if you decided Retry-After is useful, I'd suggest rewriting the code a bit since `sleepBeforeRetry` does three things: it parses Retry-After, it cuts it to the limited range and also it sleeps.

Let's define parseRetryAfter which has only 1 responsibility, then it's up to the caller what it will do with the value. Something like:

```go
func parseRetryAfter(val string) (time.Duration, bool) {
    if n, err := strconv.ParseUint(val, 10, 64); err == nil {
        return time.Duration(val) * time.Second, true
    }
    if t, err := time.Parse(time.RFC1123, val); err == nil {
        return time.Until(t), true
    }
    return 0, false
}
```

And then you can use it:

```go
if d, ok := parseRetryAfter(resp.Header.Get("Retry-After")); ok {
    time.Sleep(d)
} else {
    time.Sleep(retryBackoffDuration)
}
```

To me it's more clear and the intention is obvious, the previous version made me think a little bit more than usual.

Also it would be good to support cancellation via `req.Context().Done()`, so time.Sleep does not block e.g. if user pressed CTRL+C and plugin was terminated making the context to be cancelled.

Reply:
done